### ISSUE TITLE
Can we fine tune Gantry 5 horizontal layout to 1/10th% or to 1/100th% ?

### ISSUE THREAD

**Issue by pintobuck** [gantry/gantry5#1337]
This is the discussion so far in the RocketTheme.com forum.

Pintobuck said:
    Gantry 5 layout can adjust particles to a horizontal size in whole 1% increments. That equals about 1/8th of an inch on my page, which leaves some things out of line vertically. It would be very nice to make horizontal adjustments down to 1/10th of a %. Is there a way to do that? When I manually entered a decimal (e.g. 20.1%), Gantry refused to accept it. If it cannot be done now, could RT add this as a development request? 

Djamil Legato said:
We did have this implementation down to the first decimal of pertentages to have a greater set of layouts to pick from. However we had to revert and keep it limited to the 100 (90 really, 5-95) thicks because the amount of combinations possible otherwise was too big. We would have had to include a greater amount of CSS code also in order to cover every combination. The difficulty on managing the layout from the admin, due to the micro resizing involved, was very confusing and hard.

It would be nice if you could open this as an issue in GitHub ( github.com/gantry/gantry5 ) and bring it up to everyone's else attention. Maybe it can be discussed there and if we see that this is a huge request feature we can bring it back to the table.

I personally always preferred the decimal. It solves a great amount of issues elsewhere too, because right now we are handling a few cases, like 33.3, 16.6, 14.2, 12.5..., as special and leave everything else integer, which I never personally felt very consistent.
On the other hand I also recognize that the micro-tuning would be very cumbersome and clumsy, the usual use case is to always use whole numbers and with the decimals integration it was very hard to reach the correct % due to the many extra steps you now have in between.

Are you trying to apply this to the whole site or are you seeing yourself needing this in just a very specific case scenario?
Also note that you can always tweak this behavior on frontend with some CSS, in case you are interested in just that let us know and we can provide extra help for it.

Pintobuck said I'm applying to a Gantry 5 template for the whole site.

**Comment by w00fz** [comment]
Thanks for opening the issue in here @pintobuck

**Comment by mahagr** [comment]
The main issue why we decided to go for full percentages only (with few exceptions of 1/3, 1/6 etc) is the exponential number of CSS rules needed to make it to work. Right now we have 97 rules, but with 0.1% resolution we would need at least 900 rules. We thought about dynamically creating only the needed rules, but that doesn't work either as some of the blocks may collapse and widths would be dynamically recalculated.

In the future when all the supported browsers support flexbox (eg. when we drop support for IE9 and Android 4.4 browser), we can easily drop percentages and take full advantage of flex. Unfortunately neither of them seems to be going away for a while...

**Comment by mahagr** [comment]
One option would be to use dynamic flex rules in all supported browsers and then approximate grid for browsers that do not support flex. This would mean that layout would look misaligned in IE9 and Android 4, but we wouldn't have artificial percentages in any of the other browsers.

At the same time I would change widths from percentages into proportions. That would still allow people to use percentages, for example: 20 | 60 | 20, but would also allow 1 | 3 | 1, which works much better in layouts than percentages.

**Comment by pintobuck** [comment]
Wow, this discussion is above my head.  I hope I'm not going to make a silly suggestion, but here goes.

Some thoughts/suggestions from a nonprogrammer (me):

1)It would be good to be able to adjust Gantry 5 horizontal layout to within 1/100ths of a percent (e.g. 20.33). Many computer screens are high definition, quite large, and they will show minor misalignments in the template layout.

2) That would require 10,000 classes/rules in a style sheet (100% x 100 decimal settings), which may be impractical.

3) A Google search shows that some people put their classes into a database, which seems like a good solution.

4) The time delay to access classes in the Joomla database could make the Gantry 5 drag and drop feature frustratingly slow or impractical.

5) You could keep the Gantry 5 drag and drop feature the same as it is, with whole 1% choices from 1% - 100% (or typically from 5% - 95%).

6) Allow decimal settings (e.g. 20.33) to be put in manually. Then when users click the apply button, Gantry 5 will have time to access the database containing the 10,000 classes/rules, and users will be expecting the minor time delay for results to show on screen.

**Comment by mahagr** [comment]
Yeah, this is never-ending discussion when it gets to percentages. Which is why I suggested another approach, which would fix all the issues -- at least while the browser supports flex.

**Comment by w00fz** [comment]
One of the discussions we had internally regarding Layout Manager, was to rewrite it completely, and by that I mean redoing the UI of it.
This is by no means anything that will happen any time soon, just some planning we did ahead.
Perhaps at when we will get there we could think of reimplementing back the fractional %, I personally think that is much better than proportions, since it takes you closer to the Flex Box system and because it allows for a better fine-tuning.

5% - 95% with 1% step is already a HUGE step forward in the CMS world and I think this will satisfy the majority of the devs and users for quite some time.

However, even though no one is really feeling claustrophobic with the 1% step Layout Manager provides, I think sooner or later we will have to touch @pintobuck's point again and provide a solution for fractional steps. Perhaps some sort of opt-in setting where you can pick either or.
